yb-voyager/cmd/importData.go

```diff
@@ -241,43 +241,69 @@ func discoverFilesToImport() []*ImportFileTask {
 
 func applyTableListFilter(importFileTasks []*ImportFileTask) []*ImportFileTask {
 	result := []*ImportFileTask{}
-	includeList := utils.CsvStringToSlice(tconf.TableList)
-	log.Infof("includeList: %v", includeList)
-	excludeList := utils.CsvStringToSlice(tconf.ExcludeTableList)
-	log.Infof("excludeList: %v", excludeList)
+
+	migInfo, err := LoadMigInfo(exportDir) //TODO: handle with msr.SourceDBConf
+	if err != nil {
+		utils.ErrExit("failed to load migration info: %s", err)
+	}
+	source.DBType = migInfo.SourceDBType
+	source.Schema = migInfo.SourceDBSchema
 
 	//TODO: handle with case sensitivity later
-	standardizeCaseInsensitiveTableNames := func(tableName string) string {
+	standardizeCaseInsensitiveTableNames := func(tableName string, defaultSourceSchema string) string {
 		parts := strings.Split(tableName, ".")
 		tableName = parts[len(parts)-1]
 		if !utils.IsQuotedString(tableName) {
 			tableName = strings.ToLower(tableName)
 		}
-
 		if len(parts) > 1 {
-			migInfo := ExtractMetaInfo(exportDir) //TODO: handle with msr.SourceDBConf 
-			source.DBType = migInfo.SourceDBType
-			if parts[0] == getDefaultSourceSchemaName() {
+			if parts[0] == defaultSourceSchema {
 				return tableName
 			}
 			return fmt.Sprintf(`%s.%s`, parts[0], tableName)
 		}
 		return tableName
 	}
 
-	includeList = lo.Map(includeList, func(tableName string, _ int) string {
-		return standardizeCaseInsensitiveTableNames(tableName)
-	})
-	excludeList = lo.Map(excludeList, func(tableName string, _ int) string {
-		return standardizeCaseInsensitiveTableNames(tableName)
-	})
-
 	allTables := lo.Map(importFileTasks, func(task *ImportFileTask, _ int) string {
-		return standardizeCaseInsensitiveTableNames(task.TableName)
+		defaultSourceSchema, _ := getDefaultSourceSchemaName() // err can be ignored as these table names will be qualified for non-public schemas
+		return standardizeCaseInsensitiveTableNames(task.TableName, defaultSourceSchema)
 	})
 	slices.Sort(allTables)
 	log.Infof("allTables: %v", allTables)
 
+	flattenTables := func(pattern string) []string {
+		result := lo.Filter(allTables, func(tableName string, _ int) bool {
+			matched, _ := filepath.Match(pattern, tableName)
+			return matched
+		})
+		return result
+	}
+
+	extractTableList := func(flagTableList, listName string) []string {
+		tableList := utils.CsvStringToSlice(flagTableList)
+		var result []string
+		defaultSourceSchema, err := getDefaultSourceSchemaName()
+		var unqualifiedTables []string
+		for _, table := range tableList {
+			if err != nil && len(strings.Split(table, ".")) == 1 {
+				unqualifiedTables = append(unqualifiedTables, table)
+				continue
+			}
+			table = standardizeCaseInsensitiveTableNames(table, defaultSourceSchema)
+			result = append(result, flattenTables(table)...)
+		}
+		if len(unqualifiedTables) > 0 {
+			utils.PrintAndLog("Error: can not qualify table names %v in the %s list. \n%s", unqualifiedTables, listName, err.Error())
```

Comment:
`"qualify following names from the %s list with schema-name: %s", listName, unqualifiedTables`